This patch is based on openembedded's crosscompilation patch. Original is
(C) by Michael 'Mickey' Lauer <mlauer@vanille-media.de>

Index: Python-2.7.2/Makefile.pre.in
===================================================================
--- Python-2.7.2.orig/Makefile.pre.in	2011-06-11 16:46:26.000000000 +0100
+++ Python-2.7.2/Makefile.pre.in	2012-03-24 17:35:44.953430355 +0000
@@ -214,7 +214,7 @@
 ##########################################################################
 # Parser
 PGEN=		Parser/pgen$(EXE)
-
+HOSTPGEN=	$(PGEN)$(EXE)
 POBJS=		\
 		Parser/acceler.o \
 		Parser/grammar1.o \
@@ -540,9 +540,9 @@
 
 # Use a stamp file to prevent make -j invoking pgen twice
 $(GRAMMAR_H) $(GRAMMAR_C): Parser/pgen.stamp
-Parser/pgen.stamp: $(PGEN) $(GRAMMAR_INPUT)
+Parser/pgen.stamp: $(HOSTPGEN) $(GRAMMAR_INPUT)
 		-@$(INSTALL) -d Include
-		$(PGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
+		$(HOSTPGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
 		-touch Parser/pgen.stamp
 
 $(PGEN):	$(PGENOBJS)
Index: Python-2.7.2/configure.in
===================================================================
--- Python-2.7.2.orig/configure.in	2011-06-11 16:46:28.000000000 +0100
+++ Python-2.7.2/configure.in	2012-03-24 17:41:52.300426587 +0000
@@ -4208,12 +4208,12 @@
   ]]])],
   [ac_cv_have_long_long_format=yes],
   [ac_cv_have_long_long_format=no],
-  [ac_cv_have_long_long_format=no])
+  [ac_cv_have_long_long_format="cross -- assuming yes"])
   )
   AC_MSG_RESULT($ac_cv_have_long_long_format)
 fi
 
-if test "$ac_cv_have_long_long_format" = yes
+if test "$ac_cv_have_long_long_format" != no
 then
   AC_DEFINE(PY_FORMAT_LONG_LONG, "ll",
   [Define to printf format modifier for long long type])
